QRCoder/BitmapByteQRCode.cs

```diff
@@ -53,54 +55,85 @@ public byte[] GetGraphic(int pixelsPerModule, byte[] darkColorRgb, byte[] lightC
     {
         var sideLength = QrCodeData.ModuleMatrix.Count * pixelsPerModule;
 
-        var moduleDark = darkColorRgb.Reverse();
-        var moduleLight = lightColorRgb.Reverse();
+        // Pre-calculate color/module bytes
+        byte[] moduleDark = new byte[pixelsPerModule * 3];
+        byte[] moduleLight = new byte[pixelsPerModule * 3];
+        for (int i = 0; i < pixelsPerModule * 3; i += 3)
+        {
+            moduleDark[i] = darkColorRgb[2];
+            moduleDark[i + 1] = darkColorRgb[1];
+            moduleDark[i + 2] = darkColorRgb[0];
+            moduleLight[i] = lightColorRgb[2];
+            moduleLight[i + 1] = lightColorRgb[1];
+            moduleLight[i + 2] = lightColorRgb[0];
+        }
+
+        // Pre-calculate padding bytes
+        var paddingLen = sideLength % 4;
+
+        // Calculate filesize (header + pixel data + padding)
+        var fileSize = 54 + (3 * (sideLength * sideLength)) + (sideLength * paddingLen);
+
+        // Bitmap container
+        byte[] bmp = new byte[fileSize];
+        int ix = 0;
+
+        // Header part 1
+        Array.Copy(_bitmapHeaderPart1, 0, bmp, ix, _bitmapHeaderPart1.Length);
+        ix += _bitmapHeaderPart1.Length;
+
+        // Filesize
+        CopyIntAs4ByteToArray(fileSize, ix, ref bmp);
+        ix += 4;
 
-        var bmp = new List<byte>();
+        // Header part 2
+        Array.Copy(_bitmapHeaderPart2, 0, bmp, ix, _bitmapHeaderPart2.Length);
+        ix += _bitmapHeaderPart2.Length;
 
-        //header
-        bmp.AddRange(new byte[] { 0x42, 0x4D, 0x4C, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x36, 0x00, 0x00, 0x00, 0x28, 0x00, 0x00, 0x00 });
+        // Width
+        CopyIntAs4ByteToArray(sideLength, ix, ref bmp);
+        ix += 4;
+        // Height
+        CopyIntAs4ByteToArray(sideLength, ix, ref bmp);
+        ix += 4;
 
-        //width
-        bmp.AddRange(IntTo4Byte(sideLength));
-        //height
-        bmp.AddRange(IntTo4Byte(sideLength));
+        // Header end
+        Array.Copy(_bitmapHeaderPartEnd, 0, bmp, ix, _bitmapHeaderPartEnd.Length);
+        ix += _bitmapHeaderPartEnd.Length;
 
-        //header end
-        bmp.AddRange(new byte[] { 0x01, 0x00, 0x18, 0x00 });
-        bmp.AddRange(new byte[24]);
+        // Add header null-bytes
+        ix += 24;
 
-        //draw qr code
+
+        // Draw qr code
+        var group = new byte[(int)(sideLength / pixelsPerModule) * moduleDark.Length];
```

Comment:
```suggestion
```
`group` isn't read anymore and can be removed

Reply:
Had this removed already, but re-added it for debugging. Then forget to finally remove it... 
Fixed in: https://github.com/codebude/QRCoder/commit/86c2e47561ef35b56f4be8f70d93c2ffd5ef40c9